Cannot include dependencies with add_directories

This is the same issue that #15 attempted to solve. In CMake, we don't want to search for the dependencies (pybind11 and nlohmann_json) using `find_package` if the user is adding them using `add_directories`. In this case, the `find_package` command would fail because it would force CMake to look for these dependencies in the wrong places. So #15 added the following guards: only call `find_package(nlohmann_json)` if the target `nlohmann_json` doesn't already exist, and only call `find_package(pybind11)` if the target `pybind11 ` doesn't already exist:
```
if (NOT TARGET pybind11)
find_package(pybind11 ${pybind11_REQUIRED_VERSION} REQUIRED)
endif()
if (NOT TARGET nlohmann_json)
find_package(nlohmann_json ${nlohmann_json_REQUIRED_VERSION} REQUIRED)
endif()
```

Unfortunately, the first guard does not work. The problem is that the target `pybind11` does not exist _even if_ pybind11 has been properly added. I was using `add_directories` to add pybind11 to my project and getting the following error from CMake:

```
CMake Error at build/_deps/pybind11_json-src/CMakeLists.txt:26 (find_package):
  By not providing "Findpybind11.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "pybind11",
  but CMake did not find one.

  Could not find a package configuration file provided by "pybind11"
  (requested version 2.2.4) with any of the following names:

    pybind11Config.cmake
    pybind11-config.cmake

  Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
  "pybind11_DIR" to a directory containing one of the above files.  If
  "pybind11" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring incomplete, errors occurred!
```
This is because pybind11_json/CMakeLists.txt:26 is calling `find_package(pybind11)` when it shouldn't be, which happens because the target `pybind11` does not exist.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,7 +22,7 @@ set(${PROJECT_NAME}_VERSION ${${PROJECT_NAME}_VERSION_MAJOR}.${${PROJECT_NAME}_V
 set(pybind11_REQUIRED_VERSION 2.2.4)
 set(nlohmann_json_REQUIRED_VERSION 3.2.0)
 
-if (NOT TARGET pybind11)
+if (NOT TARGET pybind11::headers)
 find_package(pybind11 ${pybind11_REQUIRED_VERSION} REQUIRED)
 endif()
 if (NOT TARGET nlohmann_json)

